### PR TITLE
Fix CMIC CRE product in nwcsaf yaml reader

### PR DESCRIPTION
**Pull request by ninahakansson**
PPS-2021 CMIC dataset cmic_cre correspond to PPS-2018 CPP dataset cpp_reff.

 - [ ] Closes #xxxx 
 - [x] Tests added 
 - [ ] Fully documented 
 - [ ] Add your name to `AUTHORS.md` if not there already

### CODE REVIEW

File: satpy/readers/nwcsaf_nc.py [hunk 1]
```diff
@@ -140,19 +140,25 @@ def get_dataset(self, dsid, info):
                 return lat
 
         logger.debug('Reading %s.', dsid_name)
-        file_key = self._get_filekey(dsid_name, info)
-        variable = self.nc[file_key]
+        file_keys = self._get_filekeys(dsid_name, info)
+        try:
+            variable = self.nc[file_keys[0]]
+        except KeyError:
+            variable = self.nc[file_keys[1]]
```
Comment (mraspaud) [inline satpy/readers/nwcsaf_nc.py:147]: Should there be a loop here, as we can't be sure there will be only 2 items?

File: satpy/readers/nwcsaf_nc.py [hunk 2]
```diff
@@ -217,7 +223,13 @@ def _mask_variable(variable):
...
     def _prepare_variable_for_palette(self, variable, info):
         try:
-            so_dataset = self.nc[self.file_key_prefix + info['scale_offset_dataset']]
+            if not isinstance(info['scale_offset_dataset'], list):
+                so_dataset = self.nc[self.file_key_prefix + info['scale_offset_dataset']]
+            else:
+                try:
+                    so_dataset = self.nc[self.file_key_prefix + info['scale_offset_dataset'][0]]
+                except KeyError:
+                    so_dataset = self.nc[self.file_key_prefix + info['scale_offset_dataset'][1]]
```
Comment (mraspaud) [inline satpy/readers/nwcsaf_nc.py:232]: This looks very similar to the code above, could these lines be factorised together somehow?

File: satpy/etc/readers/nwcsaf-pps_nc.yaml [hunk 3]
```diff
@@ -231,17 +231,29 @@ datasets:
...
+  #  file_key: reff
+  #  file_type: [nc_nwcsaf_cmic, nc_nwcsaf_cpp]
+  #  coordinates: [lon, lat]
+
+  #cpp_reff_pal:
+  #  name: cmic_reff_pal
+  #  file_key: reff_pal
+  #  scale_offset_dataset: reff
+  #  file_type: [nc_nwcsaf_cmic]
+
```
Comment (mraspaud) [inline satpy/etc/readers/nwcsaf-pps_nc.yaml:245]: This can be removed

**Review by mraspaud — CHANGES_REQUESTED** [review]
Looks like this logic will work, nice job! Tests are missing though :)

**Review by mraspaud — CHANGES_REQUESTED** [review]
Looks good, now just tests missing.

File: satpy/tests/reader_tests/test_nwcsaf_nc.py [hunk 4]
```diff
@@ -226,6 +226,32 @@ def test_get_dataset_uses_file_key_if_present(self):
...
+        """Test that get_dataset() uses a file_key list if present."""
+        dsid_cpp = {'name': 'cpp_reff'}
+        dsid_cmic = {'name': 'cmic_cre'}
+        scale = 4
+        offset = 8
+        the_array = xr.DataArray(np.ones((5, 10)), attrs={"scale_factor": np.array(scale, dtype=float),
+                                                          "add_offset": np.array(offset, dtype=float)})
+
+        self.fh.nc["cpp_reff"] = the_array
+        self.fh.nc["cmic_cre"] = the_array
```
Comment (mraspaud) [inline satpy/tests/reader_tests/test_nwcsaf_nc.py:239]:
```suggestion
        data_array = xr.DataArray(np.ones((5, 10)), attrs={"scale_factor": np.array(scale, dtype=float),
                                                           "add_offset": np.array(offset, dtype=float)})

        self.fh.nc["cpp_reff"] = data_array
        self.fh.nc["cmic_cre"] = data_array
```

**Review by mraspaud — APPROVED** [review]
LGTM, just one naming suggestion

**Review by adybbroe — COMMENTED** [review]
I just have one question. LGTM otherwise

**Comment by adybbroe** [comment]
@mraspaud I accidentally requested you to re-review this. But now that I bothered you,p what should we do with the failed tests above? They don't seem to have anything to with the PR here? Perhaps you need to merge latest main into your PR branch @ninahakansson ?

**Comment by mraspaud** [comment]
These errors seem to come from third party libraries. Someone should look up that error and maybe report a bug to them if necessary to get it fixed.

**Comment by mraspaud** [comment]
I can try to have a look at it this afternoon

**Review by adybbroe — APPROVED** [review]
LGTM

**Comment by mraspaud** [comment]
I just merge main into this to fix the CI

**Pull request merged**